From f3c1d9a12582afe6153544809438b97a5c7c937b Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <roman.o.stratiienko@globallogic.com>
Date: Sat, 21 May 2022 13:06:51 +0300
Subject: [PATCH 1/4] egl: android: Dequeue buffer for the next frame soon
 after enqueuing

Insignificant optimization that helps to reduce input-to-draw latency.

On slow Allwinner A64 SOC this commit helps to save just a ~140uS of
frame time. Not much but still it's something.

Signed-off-by: Roman Stratiienko <roman.o.stratiienko@globallogic.com>
---
 src/egl/drivers/dri2/platform_android.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/egl/drivers/dri2/platform_android.c b/src/egl/drivers/dri2/platform_android.c
index 46a392bc809..fbbd116f2f2 100644
--- a/src/egl/drivers/dri2/platform_android.c
+++ b/src/egl/drivers/dri2/platform_android.c
@@ -1022,6 +1022,12 @@ droid_swap_buffers(_EGLDisplay *disp, _EGLSurface *draw)
       draw->ActiveRenderBuffer = draw->RequestedRenderBuffer;
    }
 
+   /* Optimization: Dequeue the buffer for a next frame:
+    * This reduces CPU time spent by next frame on a road
+    * from start-of-frame event to GPU job-submit event.
+    */
+   update_buffers(dri2_surf);
+
    return EGL_TRUE;
 }
 
-- 
2.37.2

